shared/chat/conversation/info-panel/notifications/container.js

```diff
@@ -25,67 +48,32 @@ const serverStateToProps = (notifications: Types.NotificationsState, type: 'desk
   return 'never'
 }
 
-type _StateProps = {
-  channelWide: boolean,
-  conversationIDKey: string,
-  desktop: Types.NotifyType,
-  mobile: Types.NotifyType,
-  muted: boolean,
-  saveState: Types.NotificationSaveState,
-}
-
-// Use conversationIDKey as a signal for whether StateProps is empty.
-type StateProps = _StateProps | {conversationIDKey: void}
-
-const mapStateToProps = (state: TypedState): StateProps => {
-  const conversationIDKey = Constants.getSelectedConversation(state)
-  if (!conversationIDKey) {
-    logger.warn('no selected conversation')
-    return {conversationIDKey: undefined}
-  }
+const mapStateToProps = (state: TypedState, {conversationIDKey}: OwnProps) => {
   const inbox = Constants.getSelectedInbox(state)
-  if (!inbox) {
-    logger.warn('no selected inbox')
-    return {conversationIDKey: undefined}
-  }
+  if (!inbox) throw new Error('Impossible')
```

Comment:
this code is going away soon so we can leave this but in general i think this is an anti pattern. It's very possible the inbox can be null. if there is no selection or there's some race this can happen, and since this is a deeply nested component thats connected there's no real guarantees it'll be non null. 

In my chat pr this is enforce as the selected conversation is passed down and must always exist for the children

Reply:
an example of this is the attachment popup. it assumed it was impossible for the attachment to not be there but you could delete it from the popup and that would happen before the navigation happened and you'd crash out. instead its safer to just render nothing

Reply:
Also confused here -- the old code avoids this problem entirely. Should we then just leave it?

Reply:
Per chat, just added comment, since it'll go away soon anyway